packages/block-editor/src/components/url-input/index.js

```diff
@@ -310,6 +326,13 @@ class URLInput extends Component {
 		}
 	}
 
+	reset() {
+		// move focus to the input
+		this.inputRef.current.focus();
+		// reset our suggestions
+		this.updateSuggestions();
```

Comment:
Do we need to call `this.updateSuggestions()` here given that it is called in `onFocus` or do you feel that `reset` shouldn't be aware of this fact?

Reply:
The issue I was running into is that the suggestions were not returning to their initial state because there was a manual URL suggestion hanging around. You can see the same effect on master by typing a URL-like string then clicking the reset x.

Reply:
I tested this out again, and it seems that we need to forcefully call `this.updateSuggestions()` to get it to work correctly.

The other option was to do `this.setState` manually to reset the suggestions, but I was worried about the impact of manually setting that.

Reply:
> we need to forcefully call this.updateSuggestions() to get it to work correctly.

Isn't that a little odd then? Because the `this.inputRef.current.focus();` above also ends up triggering `this.updateSuggestions` via the `onFocus` handler. 

I wonder whether it would be better to just manually reset the suggestions here rather than relying on a side effect of calling `updateSuggestions` without a value. At least that way we're being explicit about our intent (ie: if the input has no value on update then clear all suggestions).

I'd also value an opinion from @aduth here.

Reply:
https://github.com/WordPress/gutenberg/pull/19850/files#diff-7f80e90a821240f277aae4a7d2d9e778R88-R91

The focus only fires `this.updateSuggestions` if there is an input value present. Loads of tests were failing by forcefully calling it anytime there was a focus.

Reply:
Doing something like 
```
	reset() {
		// move focus to the input
		this.inputRef.current.focus();
		// reset our suggestions
		this.setState({
			suggestions: [],
			selectedSuggestion: null,
		});
	}
```

does work, but the tests I wrote fail. I can take a look at fixing those i you feel like this is a better way to go.

Reply:
This doesn't look relevant any more right @jeryj cos `reset` got removed?

Reply:
Yup! Can be resolved IMO

Reply:
I removed the entire `reset()` function just a bit ago